thrift.js

```diff
@@ -65,14 +82,31 @@ function Thrift(options) {
     self.exceptions = Object.create(null);
     self.unions = Object.create(null);
     self.typedefs = Object.create(null);
+    self.modulesByName = Object.create(null);
+    self.modulesByPath = options.modulesByPath || Object.create(null);
+    self.linked = false;
+    self.filepathThriftMemo = options.filepathThriftMemo || Object.create(null);
```

Comment:
Technically yes, except for the latter is more like a cache of all files threaded through and the former is only the modules relevant to the current Thrift instance.